### PR TITLE
prevent KeyError: 'purelib' 

### PR DESCRIPTION
**Pull request by pawl**
This might be another potential fix for the issues in: https://github.com/celery/kombu/pull/1462

Currently we're seeing this error:
```
Traceback (most recent call last):
File "setup.py", line 60, in
scheme['data'] = scheme['purelib']
KeyError: 'purelib'
```

This seems to describe what the code with the error is doing: https://stackoverflow.com/a/3042436

That code was added in the initial commit for kombu and `data_files` is no longer defined anymore. So, maybe the hack to copy data files to the module directory isn't necessary anymore? Or maybe this should be handled in `MANIFEST.in` instead?

### CODE REVIEW

**Comment by pawl** [comment]
The root cause of this issue seems to be this commit: https://github.com/pypa/setuptools/commit/f32af4e73a3f994f3a76898bb404919c5722508d

That new scheme that got added didn't have purelib defined.

**Comment by auvipy** [comment]
I was thinking of removing this as well but should we open an issue in setup tolls as well?

**Comment by auvipy** [comment]
https://github.com/celery/kombu/pull/1466/files doesn't seem to work

**Comment by pawl** [comment]
@auvipy I created an issue with setuptools here: https://github.com/pypa/setuptools/issues/2936

**Comment by pawl** [comment]
@auvipy the setuptools maintainer suggests:
> My preference would be for the hack to adjust its expectations or to remove the hack altogether if possible, but I realize that may not be possible in all scenarios.

So maybe we should go ahead with this?

We could also potentially compare the files from the packages built before and after this change to be extra careful.

**Comment by auvipy** [comment]
what about using 

try adding this to your pip install:

SETUPTOOLS_USE_DISTUTILS=stdlib pip install .... 

As mentioned from setuptools issue

**Comment by auvipy** [comment]
I will also come back to PR, but need some more careful analysis before merging this :D

**Review by matusvalo — APPROVED** [review]
Based what I read in https://stackoverflow.com/questions/1612733/including-non-python-files-with-setup-py/3042436#3042436 we should be fine since we are not using `data_files` as mentioned in the link right? But to be honest this topic is not my expertise :-(

**Comment by auvipy** [comment]
> Based what I read in https://stackoverflow.com/questions/1612733/including-non-python-files-with-setup-py/3042436#3042436 we should be fine since we are not using `data_files` as mentioned in the link right? But to be honest this topic is not my expertise :-(

no worries. I have been studying these type of topics  for last few weeks.

**Comment by pawl** [comment]
@matusvalo @auvipy I just compared the contents of `site-packages/kombu-5.2.3-py3.8.egg/` before and after this change (with the most recent changes from master), and I confirmed this change doesn't affect which files end up in the package. I also created a fresh virtualenv for each `python setup.py install`. 

Here's the output I was looking at: https://gist.github.com/pawl/4828c02abba078a154bf7b77f22ba0dd

So, I think that confirms the lines this PR removes are no longer necessary since we're not using `data_files`.

**Review by auvipy — APPROVED** [review]

**Comment by auvipy** [comment]
thank you @pawl

**Pull request merged**
